File: src/main/java/dev/vrba/simp/command/fun/animals/endpoints/DogsApiEndpoint.java
package dev.vrba.simp.command.fun.animals.endpoints;

import dev.vrba.simp.command.fun.animals.AnimalApiEndpoint;
import org.jetbrains.annotations.NotNull;
import reactor.core.publisher.Mono;

import java.io.InputStream;
import java.util.Set;

public class DogsApiEndpoint extends AnimalApiEndpoint {

    @Override
    @NotNull
    public Set<String> getNames() {
        return Set.of("dog", "doggo", "doge", "dogs", "doggos", "doges", "cheems", "cheemsburger", "pupper");
    }

    @Override
    @NotNull
    public String getUrl() {
        return "https://shibe.online/api/shibes?count=1";
    }

    @Override
    @NotNull
    public String getTitle() {
        return "Woof";
    }

    @Override
    @NotNull
    public Mono<String> extractImageFromResponse(@NotNull Mono<InputStream> response) {
        return this.extractImageFromJson(response, json -> json.get(0).asText());
    }
}

File: Services/Endpoints/DogsApiEndpoint.cs
using System.Net.Http.Json;

namespace SimpBot.Services.Endpoints;

public class DogsApiEndpoint : ImageApiEndpoint
{
    public override IEnumerable<string> Names { get; } = new[]
        {"dog", "doggo", "doge", "dogs", "doggos", "doges", "cheems", "cheemsburger", "pupper", "puppers"};

    // TODO: Optimize per-endpoint fetching -> return list of urls
    public override string Url => "https://shibe.online/api/shibes?count=1";

    public override async Task<string?> ExtractImageUrlAsync(HttpResponseMessage response)
    {
        var data = await response.Content.ReadFromJsonAsync<List<string>>();
        var image = data?[0];

        return image;
    }
}